### PR TITLE
revising security section

### PR DESCRIPTION
**Pull request by aleks-mariusz**
Went through the security-related sections in the docs, changed references from just "authorization" to proper feature-name "PX-Security", also enhanced/elaborated on certain parts to be more clear and consistent.

### CODE REVIEW

File: content/concepts/authorization/overview.md [hunk 1]
```diff
@@ -1,107 +1,203 @@
...
+
+Once Portworx is able to determine the token to be validated was signed by the TA it
+is configured to trust, Portworx, by extension, is also able to trust the contents of
+the token (due to the user of cryptographic signatures that are part of the token),
+and it then accepts the identity of the user providing the token.
+
+### Sources
+
+Portworx supports two sources for token generation: either via the industry-standard
+_OpenID Connect_ and using _self-generated_ tokens.
```
Comment (lpabon) [inline content/concepts/authorization/overview.md]: should this be 'or' instead of 'and'?

Comment (aleks-mariusz) [reply content/concepts/authorization/overview.md]: I wondered the same thing, but then elsewhere in the docs, it mentions it supports both at the same time (i am guessing that the internal stork communication uses a self-generated token, irrespective if you also set up OpenID Connect, so i would presume it have the ability to use both simultaneously?

File: content/concepts/authorization/install.md [hunk 2]
```diff
@@ -1,62 +1,79 @@
...
+description: Explains how to enable PX-Security in Portworx
 keywords: portworx, security, ownership, tls, rbac, claims, jwt, oidc
 weight: 30
 series: authorization
 ---
 
-This page will give an overview of how to enable authorization in _Portworx_. For specific details on how to enable authorization in _Portworx_ for a given container orchestration system, please refer to the appropriate installation instructions in this site.
+This page will discuss enabling PX-Security in Portworx. 
+
+For specific details on enabling PX-Security in _Portworx_ for a specific container
```
Comment (nathan-px) [inline content/concepts/authorization/install.md]:
```suggestion
For specific details on enabling PX-Security in Portworx for a specific container
```

File: content/concepts/authorization/install.md [hunk 3]
```diff
@@ -1,62 +1,79 @@
...
+orchestration system, please refer to the relevant installation instructions on this
+site for the given orchestration system.
 
 ## Requirements
-To enable authorization in _Portworx_, administrators must provide the following
-information:
+
+To enable PX-Security in _Portworx_, administrators must provide the
```
Comment (nathan-px) [inline content/concepts/authorization/install.md]:
```suggestion
To enable PX-Security in Portworx, administrators must provide the 
```

File: content/concepts/authorization/install.md [hunk 4]
```diff
@@ -1,62 +1,79 @@
...
+  * OIDC Issuer
+  * OIDC Client id
+
+If self-signed token model is used, then the following is required:
+  * JWT issuer
+  * JWT shared secret or JWT public key
+
+## Setup Parameters
+
+The parameters above are required and must be provided to _Portworx_ to enable
```
Comment (nathan-px) [inline content/concepts/authorization/install.md]:
```suggestion
The parameters above are required and must be provided to Portworx to enable
```

File: content/concepts/authorization/overview.md [hunk 5]
```diff
@@ -1,107 +1,203 @@
...
+title: Security Overview
+description: Security concepts and features available within Portworx
 keywords: portworx, security, ownership, tls, rbac, claims, jwt, oidc
 weight: 10
 series: authorization
 ---
+## Introduction
 
-From version 2.1, Portworx has support for security. This document provides a high-level overview of the different components used to secure Portworx.
+Portworx has enhanced authN/authZ security as of version 2.1 called PX-Security. This
```
Comment (nathan-px) [inline content/concepts/authorization/overview.md]:
```suggestion
As of version 2.1, Portworx features enhanced authN/authZ security called PX-Security. This
```

File: content/concepts/authorization/overview.md [hunk 6]
```diff
@@ -1,107 +1,203 @@
...
 
-This secure model enables Portworx to only have to verify the validity of the token to authenticate the user. Portworx then destroys the token ensuring **tokens are never saved on a Portworx system**.
+Our PX-Security feature centers around the ubiquitous JWT-based authentication and
+authorization mechanisms. These security techniques are leveraged by many IT systems
+and are a proven model for user/account identification and permissioning.
 
-The token contains a section called _claims_ which not only identifies the user but also provides authorization information in the form of RBAC. Portworx uses the RBAC information to determine if the user is authorized to make the request.
+Within a Portworx Cluster with PX-Security enabled, a storage-client entity (user)
+first authenticates with a token authority (TA), which generates a token for the
+user , signed using either a private key or a shared secret. The user then provides
```
Comment (nathan-px) [inline content/concepts/authorization/overview.md]:
```suggestion
user, signed using either a private key or a shared secret. The user then provides
```

File: content/concepts/authorization/overview.md [hunk 7]
```diff
@@ -1,107 +1,203 @@
...
+this token to Portworx for identification. **No passwords are ever transmitted to
+Portworx**.
+
+This approach allows Portworx to only have to verify the validity of a token issued
+to a user for Portworx to be able to authenticate the user (as it is configured to
+trust the private key/shared key only the TA would also have).  Portworx discards
```
Comment (nathan-px) [inline content/concepts/authorization/overview.md]:
```suggestion
trust the private key/shared key only the TA would also have). Portworx discards
```

File: content/concepts/authorization/overview.md [hunk 8]
```diff
@@ -1,107 +1,203 @@
...
+permitted to make their storage-related request.
 
 ## Terminology
 
 | Term | Definition |
 | ---- | ---------- |
 | ACL | Access Control List identifying those accounts which can access a resource|
 | claims | Information contained in the payload of the JWT identifying the owner of the request |
-| CO | Container Orchestration System like Kubernetes, Mesosphere, or Nomad |
+| CO | Container Orchestrator like Kubernetes, Mesosphere, Nomad, etc |
```
Comment (nathan-px) [inline content/concepts/authorization/overview.md]:
```suggestion
| CO | Container Orchestrator like Kubernetes, Mesosphere, Nomad, etc. |
```

File: content/concepts/authorization/overview.md [hunk 9]
```diff
@@ -1,107 +1,203 @@
...
+like Active Directory, LDAP directories, etc.
+
+The other option is simply using self-generated tokens, where the administrator would
+manually generate a token for each user using their own TA application. For
+convenience, Portworx provides a method of generating tokens using `pxctl`.
+
+While both can be used, choosing one over the other of these options is a trade-off
+between scaleability and simplicity.  Whereas integrating with OpenID connect can be
+initially complex, it is the superior solution once in place and is enterprise grade,
+while using only the simpler self-signed token approach will requires manual
```
Comment (nathan-px) [inline content/concepts/authorization/overview.md]:
```suggestion
while using only the simpler self-signed token approach requires manual
```

File: content/concepts/authorization/overview.md [hunk 10]
```diff
@@ -1,107 +1,203 @@
...
+management of users/tokens and does not scale very well beyond a handful of
+identities.
+
+#### OpenID Connect (OIDC)
+
+With its ability to manage anywhere from tens to thousands of users, it is strongly
```
Comment (nathan-px) [inline content/concepts/authorization/overview.md]:
```suggestion
With its ability to manage anywhere from tens to thousands of users, Portworx strongly
```

File: content/concepts/authorization/overview.md [hunk 11]
```diff
@@ -1,107 +1,203 @@
...
+recommended that enterprise customers enable PX-Security with an OpenID
```
Comment (nathan-px) [inline content/concepts/authorization/overview.md]:
```suggestion
recommends that enterprise customers enable PX-Security with an OpenID
```

File: content/concepts/authorization/overview.md [hunk 12]
```diff
@@ -1,107 +1,203 @@
...
+Connect service. There are multiple public and private OIDC providers. Customers
+could use public providers such as [Okta.com](https://developer.okta.com/docs/reference/api/oidc/), [Auth0.com](https://auth0.com/docs/protocols/oidc), etc, or private provider software such as [Keycloak](https://www.keycloak.org/docs/latest/securing_apps/index.html#openid-connect-2), [Dex](https://github.com/dexidp/dex#dex---a-federated-openid-connect-provider), [CloudFoundry UAA](https://github.com/cloudfoundry/uaa#cloudfoundry-user-account-and-authentication-uaa-server), or [Tremolo Security OpenUnison](https://www.tremolosecurity.com/docs/tremolosecurity-docs/1.0.15/openunison/openunison-manual.html).
```
Comment (nathan-px) [inline content/concepts/authorization/overview.md]:
```suggestion
can use public providers such as:
* [Okta.com](https://developer.okta.com/docs/reference/api/oidc/)
* [Auth0.com](https://auth0.com/docs/protocols/oidc)

Customers can also use private provider software such as:
* [Keycloak](https://www.keycloak.org/docs/latest/securing_apps/index.html#openid-connect-2)
* [Dex](https://github.com/dexidp/dex#dex---a-federated-openid-connect-provider)
* [CloudFoundry UAA](https://github.com/cloudfoundry/uaa#cloudfoundry-user-account-and-authentication-uaa-server)
* [Tremolo Security OpenUnison](https://www.tremolosecurity.com/docs/tremolosecurity-docs/1.0.15/openunison/openunison-manual.html)
```

File: content/concepts/authorization/pre-install.md [hunk 13]
```diff
@@ -1,41 +1,46 @@
...
 weight: 20
 series: authorization
 ---
 
-To enable authorization you must first determine which type of authorization
-model you will be using. You can choose to connect to an OIDC provider or use
-self-signed tokens [Security Overview](/concepts/authorization/overview)) or
-both.
+To enable PX-Security, you must first determine which source of tokens you will be
+using. You can choose to leverage an OIDC provider or use self-signed tokens or both.
```
Comment (nathan-px) [inline content/concepts/authorization/pre-install.md]:
```suggestion
using. You can choose to leverage an OIDC provider, use self-signed tokens, or both.
```

File: content/concepts/authorization/overview.md (hunk 1 above)
Comment (nathan-px) [reply content/concepts/authorization/overview.md]: It shouldn't lead with "either" if it supports both.

File: content/concepts/authorization/overview.md [hunk 14]
```diff
@@ -1,107 +1,203 @@
...
+this is how Portworx knows to trust the TA.
+
+Once Portworx is able to determine the token to be validated was signed by the TA it
+is configured to trust, Portworx, by extension, is also able to trust the contents of
+the token (due to the user of cryptographic signatures that are part of the token),
+and it then accepts the identity of the user providing the token.
+
+### Sources
+
+Portworx supports two sources for token generation: either via the industry-standard
```
Comment (nathan-px) [inline content/concepts/authorization/overview.md]:
```suggestion
Portworx supports two sources for token generation: via the industry-standard
```

File: content/concepts/authorization/_index.md [hunk 15]
```diff
@@ -2,11 +2,11 @@
 title: Security
 weight: 1
 hidesections: true
-description: Understanding Portworx security model
+description: Understanding how PX-Security works and enabling it
```
Comment (nathan-px) [inline content/concepts/authorization/_index.md]:
```suggestion
description: Understanding and enabling PX-Security
```

**Pull request merged**